### ISSUE TITLE
Support for public holiday in opening hours parser 

### ISSUE THREAD

**Issue by HolgerJeromin** [osmandapp/Osmand#2610]
To support public holiday in the opening hours parsing we could:

* use the holiday info from https://github.com/opening-hours/opening_hours.js repository or
* add a button "today is a public holiday" or "school holiday" .

**Comment by vshcherb** [comment]
Agree, though the solution doesn't seems to be easy. I think we will better use web service rather incorporate that code into OsmAnd

**Comment by HolgerJeromin** [comment]
OsmAnd knows in which country I am. And the other repo has the data per country. I don't think supporting them would not be so much code. @ypid

**Comment by ypid** [comment]
Hey guys. Thanks for the CC. I think in the long run it would be nice to have one opening_hours implementation (with bindings for the important languages/platforms). See also https://github.com/osmandapp/Osmand/issues/473 and https://github.com/opening-hours/opening_hours.js/issues/136
Unfortunately, there is no ready to use jar lib available yet. This might still take a while. Maybe you guys can help with that?

https://github.com/osmandapp/Osmand/blob/master/OsmAnd-java/src/net/osmand/util/OpeningHoursParser.java
is becoming a more and more sophisticated with currently around 700 LOC implementation.

**Comment by HolgerJeromin** [comment]
I would love to have the reference implementation used, but now pulling the PH data out of it is probably possible.

**Comment by ypid** [comment]
@HolgerJeromin Sure, you can reuse the PH definitions. I recently [documented the data format](https://github.com/opening-hours/opening_hours.js/blob/master/docs/holidays.md) :) To do this, I think it would also be a good idea to try to provide the [Nominatim info](https://github.com/opening-hours/opening_hours.js/blob/master/docs/holidays.md#deciding-which-holidays-do-apply) but without actually querying Nominatim. OsmAnd should be able to do this on it’s own without contacting remote servers.

**Comment by HaoTian22** [comment]
maybe another option is to let user import/subscribe their ics files that contain holiday information (i believe this is available in almost all contries, released by the government or third-party developer), and parse all 'all day event' in the calendar as public holiday
